Source/Graphic/HCRaster+Draw.c

```diff
@@ -0,0 +1,605 @@
+//
+//  HCRaster+Draw.c
+//  HollowCore
+//
+//  Created by Matt Stoker on 4/18/20.
+//  Copyright © 2020 HollowCore. All rights reserved.
+//
+
+#include "HCRaster+Draw.h"
+
+//----------------------------------------------------------------------------------------------------------------------------------
+// MARK: - Curve Drawing Operations
+//----------------------------------------------------------------------------------------------------------------------------------
+void HCRasterDrawPoint(HCRasterRef self, HCReal x, HCReal y, HCColor color) {
+    HCRasterSetPixelAt(self, round(x), round(y), color);
+}
+
+void HCRasterDrawLine(HCRasterRef self, HCReal x0, HCReal y0, HCReal x1, HCReal y1, HCColor c0, HCColor c1) {
+    // Draw using direct evaluation informed by derivitive
+    HCReal dx = x1 - x0;
+    HCReal dy = y1 - y0;
+    for (HCReal t = 0.0; t <= 1.0;) {
+        HCReal tc = 1.0 - t;
+        HCReal a = tc;
+        HCReal b = t;
+        HCReal x = a * x0 + b * x1;
+        HCReal y = a * y0 + b * y1;
+        
+        HCRasterDrawPoint(self, x, y, HCColorCombine(c0, c1, t));
+        t += fmax(0.00001, 1.00000 / fmax(fabs(dx), fabs(dy)));
+    }
+    
+    // Draw using Bresenham's Algorithm
```

Comment:
Do we want these other algorithms to stay here?

Reply:
No, but using Bresenham's with a color interpolating t value made things look MUCH nicer. Thanks for reminding me!